elastic/shared/runners/remote_cluster.py

```diff
@@ -215,10 +215,13 @@ async def __call__(self, multi_es, params):
             if cluster_name in params.get("ignore-clusters", []):
                 self.logger.info(f"Multi cluster wrapped runner [{base_runner}] ignoring cluster [{cluster_name}].")
                 continue
-            runner_for_op = unwrap(runner_for(base_runner))
+            runner_for_op = runner_for(base_runner)
             self.logger.info(f"Multi cluster wrapped runner [{base_runner}] executing on cluster [{cluster_name}].")
-            # just call base runner op, don't mess with 'return' values
-            coroutines.append(runner_for_op(cluster_client, params))
+            # call original runner assuming a single-cluster one, see https://github.com/elastic/rally/pull/488
+            # and https://github.com/elastic/rally/pull/1563 for additional context
```

Comment:
Actually we cannot assume that, this requires a modification.

Reply:
I've generalized in https://github.com/elastic/rally-tracks/pull/737/commits/e3f42eab452cb9338e87d204b619433e1c7fa6d8. Today `multi-cluster-wrapper` is only used with built-in Rally runners which are exclusively single-cluster. But I can imagine a situation where someone wants to iterate through all the clusters providing a dictionary of clusters for the runner to choose from. Perhaps that's too much.